listing-rewards/program/src/assertions.rs

```diff
@@ -46,6 +46,21 @@ pub fn assert_belongs_to_rewardable_collection(
     Ok(())
 }
 
+pub fn assert_rewardable_collection_deleted_if_initialized(
+    rewardable_collection: &Account<RewardableCollection>,
+) -> Result<()> {
+    if !rewardable_collection.is_initialized {
+        return Ok(());
+    }
```

Comment:
What constitutes is_initialized? I'm thinking anchor initialize the account before our handle action is called.

Reply:
It's basically a pattern I learned from the Paulx Escrow tutorial, where he essentially had this Trait called IsInitialized which it's basically a function returning this boolean value. Since then, I got accustomed to adding this. I can remove it for sure if felt unnecessary though.